formatting/getFiles

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+bashPass="\033[32;1mPASSED -"
+bashInfo="\033[33;1mINFO -"
+bashFail="\033[31;1mFAILED -"
+bashEnd="\033[0m"
+
+# Check number of arguments
+files=""
+file=""
+excludeDirs=""
+excludeFiles=""
+includeExtensions=""
+
+# Make the assumption that Mac users will be using FreeBSD Get Opt
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    while [ $# -gt 0 ]; do
+    case "$1" in
+        -ed= | --exclude-dirs=* )
+            excludeDirs="-E $(echo "${1#*=}" | sed -E 's/,/ -E /g')"
+            shift
+            ;;
+        -ef= | --exclude-files=* )
+            excludeFiles="-E $(echo "${1#*=}" | sed -E 's/,/ -E /g')"
+            shift
+            ;;
+        -ie= | --include-extensions=* )
+            includeExtensions="-e $(echo "${1#*=}" | sed -E 's/,/ -e /g')"
+            shift
+            ;;
+        -h | --help )
+                echo -e "$bashInfo Find all .c and .h files with the Amazon copyright in them $bashEnd"
+                echo -e "$bashInfo It exports this to a bash array variable called \"files\" $bashEnd"
+                echo -e "$bashInfo This script can take in two optional arguments $bashEnd"
+                echo -e "$bashInfo -ef= | --exclude-files=:       A comma seperated list of files to exclude $bashEnd"
+                echo -e "$bashInfo -ed= | --exclude-dir=:         A comma seperated list of directories to exclude $bashEnd"
+                echo -e "$bashInfo -ie= | --include-extensions=:  Any additional exstensions to search for $bashEnd"
+                exit 0
+            ;;
+        *)
+            echo "Unknown option: $1"
+            exit 1
+            ;;
+        esac
+    done
+    # For FreeBSD (default MacOS shell) assume that you have fd
```

Comment:
FreeBSD is an OS, not a shell. You probably mean zsh?

Reply:
Also the shebang is bash so shouldn't end up running under zsh

Reply:
Was putting that there since the man page for getopt on mac states:
```
     getopts(1), sh(1), getopt(3)

HISTORY
     Written by Henry Spencer, working from a Bell Labs manual page.  Behavior believed identical to the Bell version.  Example changed in FreeBSD version 3.2 and 4.0.

BUGS
     Whatever getopt(3) has.

     Arguments containing white space or embedded shell metacharacters generally will not survive intact; this looks easy to fix but is not.  People trying to fix getopt or the example in this manpage should check the history of this file in FreeBSD.
```
Where just wanted to point out that the version mac has, by default, is the one from FreeBSD evidently